### ISSUE TITLE
Batched Changes can create cycles in components two way bindings

### ISSUE THREAD

**Issue by justinbmeyer** [bitovi/canjs#1579]
If a component might always provide a new value everytime set is called like:

```js
define: {
	// always returns a different list:
	variations: {
		set: function(variations) {
			return new can.List(variations.attr());
		}
	}
}
```

And that component is two way bound to another component:

```
<product-swatch-color variations="{variations}"></product-swatch-color>
```

If variations is changed in a batch, this will cause a never-ending series of variations changes.  

The fix is to make sure that when a scope change causes a viewModel change, the viewModel change should not update the scope.

**Comment by justinbmeyer** [comment]
@daffl Added another change for this issue.  Realized I did not implement it right.

**Comment by daffl** [comment]
Should be fully fixed now with #1590
